types/react-native/index.d.ts

```diff
@@ -4904,11 +4906,8 @@ export namespace NavigatorStatic {
 
 // @see https://github.com/facebook/react-native/blob/0.34-stable\Libraries\StyleSheet\StyleSheetTypes.js
 export namespace StyleSheet {
-
-    type Style = ViewStyle | TextStyle | ImageStyle
```

Comment:
That's removed so when you receive an error it is more meaningful. Also it's not needed since `flatten` types defined differently.